File: java/RegexPH.java
package irresponsible.crouton;

import java.util.regex.*;
import clojure.lang.IPersistentVector;
import clojure.lang.IPersistentMap;

public final class RegexPH extends PH {
  private final Pattern pat;
  public RegexPH(Object n, Pattern p, IRoute next) {
    super(n,next);
    if (null == p)
      throw new IllegalArgumentException("pattern must not be nil");
    pat = p;
  }
  public final Object match(IPersistentVector pieces, IPersistentMap places) {
    String piece = (String) pieces.nth(0, null);
    if (piece == null) return null;
    Matcher m = pat.matcher(piece);
    return m.matches() ? forward(piece, pieces, places) : null;
  }
}

File: kotlin/src/RegexPH.kt
package irresponsible.crouton

import java.util.regex.*
import clojure.lang.IPersistentVector
import clojure.lang.IPersistentMap

class RegexPH(n: Any, private val pat: Pattern, next: IRoute) : PH(n, next) {
  override fun match(pieces: IPersistentVector, places: IPersistentMap): Any? {
    val piece = pieces.nth(0, null)
    if (piece == null) return null
    val piece2 = piece as String // fucking kotlin
    val m = pat.matcher(piece)
    return if (m.matches()) forward(piece, pieces, places) else null
  }
}
